src/util.cpp

```diff
@@ -336,6 +336,11 @@ static std::string LogTimestampStr(const std::string &str, std::atomic_bool *fSt
     return strStamped;
 }
 
+bool IsStdoutTty()
+{
+    return isatty(STDOUT_FILENO);
```

Comment:
I know you said that MingW includes a compatible unistd.h version for Windows but on Visual Studio/msvc isatty is deprecated and replaced by _isatty. Also STDOUT_FILENO (1) is not defined. There's currently some PRs for msvc support so just a heads up that this will likely break things there. #11526 and #11873